### PR TITLE
613 migration to nextjs app router

### LINKED ISSUE #613: Migration to next.js app router
**Description**
To use all advantages of the app router Essencium should be refactored from pages to app router.

**Scope**
Implement app router architecture

- [ ] implement changes
- [ ] update `MIGRATION.md`
- [ ] update docs

### PR DESCRIPTION
**Pull request by PhilKsr**
## DESCRIPTION

This PR contains the migration from next.js pages router to app router.
It breaks down to following changes and follows the [migrationguide](https://nextjs.org/docs/app/building-your-application/upgrading/app-router-migration#migrating-from-pages-to-app)
- replace _document.tsx and _app.tsx with layout.tsx
- outsource and reorganize providers
- replace next/router with next/navigation and replace used router functionality with new api
- replace next-i18next with react-i18next to migrate translations
- replace all index.tsx with page.tsx
- outsource client side components to own files to prepare refactoring

Whats missing?
- overall refactoring concerning server and client side functionality and components
- refactoring of TranslationView.tsx (several bugs in there)
- adjustments to hot reload app on language change 
- documentation (will follow after technical review)

### TO-DO

- [x] implement and update tests
- [ ] update docs
- [ ] pull `main` & resolve merge conflicts
- [ ] check Vercel deployment

### CLOSES

closes #613

### CODE REVIEW

**Comment by PhilKsr** [comment]
@qhantom nx causes errors during linting. Would you have a quick look what could be the error and start a review afterwards? 🐳

File: packages/app/package.json [hunk 1]
```diff
@@ -45,8 +45,10 @@
     "dayjs": "1.11.7",
     "dotenv": "16.3.1",
     "i18next": "23.11.2",
+    "i18next-resources-to-backend": "^1.2.1",
     "jotai": "2.0.3",
     "next": "14.2.2",
+    "next-i18n-router": "^5.5.1",
     "next-i18next": "15.3.0",
```
Comment (qhantom) [inline packages/app/package.json:52]: Can this be removed since we only need `react-i18next`?

File: packages/app/src/app/[locale]/(main)/layout.tsx
Comment (qhantom) [inline packages/app/src/app/[locale]/(main)/layout.tsx:1]: It saw that the current nav item is no longer highlighted when on a route and the language is set to german

image

**Review by qhantom — CHANGES_REQUESTED** [review]
When building the app, I get the following error:

image

**Comment by qhantom** [comment]
> @qhantom nx causes errors during linting. Would you have a quick look what could be the error and start a review afterwards? 🐳

You can just check it locally by building the app on your system. The linting is executed than and throws the same error(s) as nx does

**Comment by qhantom** [comment]
For some reason, the logo at the upper left corner is flickering on route change, could you reproduce that?

File: packages/app/src/app/[locale]/(main)/layout.tsx
Comment (PhilKsr) [reply packages/app/src/app/[locale]/(main)/layout.tsx:1]: Thats something we have to fix in the lib in another PR, just like changing the locale. 
I will create a ticket for that. 👍🏼

**Comment by PhilKsr** [comment]
> For some reason, the logo at the upper left corner is flickering on route change, could you reproduce that?

Not really. If you mean the unexpected fold in Home component - thats also something for the lib.

File: packages/app/src/app/[locale]/(main)/layout.tsx [hunk 2]
```diff
@@ -17,23 +17,17 @@
...
-      <body>
-        <div id="notification" />
-        <Main />
-        <NextScript />
-      </body>
-    </Html>
+export default function MainLayout({ children }: Props): JSX.Element {
+  return (
+    <RouteProtector>
+      <AuthLayout>{children} </AuthLayout>
```
Comment (qhantom) [inline packages/app/src/app/[locale]/(main)/layout.tsx:30]: One space bro 🤜

**Review by qhantom — APPROVED** [review]
Nice one! Fits, wiggles and has air.

**Review by CathrinTruchan — COMMENTED** [review]
Nice 👯

**Pull request merged**